nixos/modules/services/networking/sunshine.nix

```
@@ -0,0 +1,148 @@
+{ config, lib, pkgs, utils, ... }:
+let
+  inherit (lib) mkEnableOption mkPackageOption mkOption mkIf types optionals getExe;
+  inherit (utils) escapeSystemdExecArgs;
+  cfg = config.services.sunshine;
+
+  # ports used are offset from a single base port, see https://docs.lizardbyte.dev/projects/sunshine/en/latest/about/advanced_usage.html#port
+  generatePorts = port: offsets: map (offset: port + offset) offsets;
+  defaultPort = 47989;
+
+  appsFormat = pkgs.formats.json { };
+  settingsFormat = pkgs.formats.keyValue { };
+
+  appsFile = appsFormat.generate "apps.json" cfg.applications;
+  configFile = settingsFormat.generate "sunshine.conf" (cfg.settings // (if cfg.applications.apps == [ ] then { } else { file_apps = appsFile; }));
+in
+{
+  options = {
+    services.sunshine = with types; {
```

nit; reduces indentation
```suggestion
  options.services.sunshine = with types; {
```

---

Yeah, good point :) done